### PR TITLE
Improve display module

### LINKED ISSUE #223: Take include and exclude parameters into account in mimebundle_repr
For now, include and exclude are ignored in this function. But we might want to respect them.

### LINKED ISSUE #238: Can't display Plotly figures in Jupyter Lab
Having trouble displaying Plotly figures in jupyter lab when running xeus-python kernel.

```
import plotly.graph_objects as go
fig = go.Figure(data=go.Bar(y=[2, 3, 1]))
fig.show()
```

results in:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
In  [1]:
Line 3:     fig.show()

File C:\Users\richa\Anaconda3\envs\analysis\lib\site-packages\plotly\basedatatypes.py, in show:
Line 2794:  return pio.show(self, *args, **kwargs)

File C:\Users\richa\Anaconda3\envs\analysis\lib\site-packages\plotly\io\_renderers.py, in show:
Line 376:   bundle = renderers._build_mime_bundle(fig_dict, renderers_string=renderer, **kwargs)

File C:\Users\richa\Anaconda3\envs\analysis\lib\site-packages\plotly\io\_renderers.py, in _build_mime_bundle:
Line 285:   self._activate_pending_renderers(cls=MimetypeRenderer)

File C:\Users\richa\Anaconda3\envs\analysis\lib\site-packages\plotly\io\_renderers.py, in _activate_pending_renderers:
Line 192:   renderer.activate()

File C:\Users\richa\Anaconda3\envs\analysis\lib\site-packages\plotly\io\_base_renderers.py, in activate:
Line 324:   ipython_display.display_html(script, raw=True)

AttributeError: module 'display' has no attribute 'display_html'
---------------------------------------------------------------------------
```

Suspect this behaviour is linked to https://github.com/jupyter-xeus/xeus-python/issues/106, but didn't know for sure.

### PR DESCRIPTION
**Pull request by martinRenou**
Should fix #223, fix #238

### CODE REVIEW

**Comment by martinRenou** [comment]
This should fix `plotly`
plotly
 support

**Comment by AntoineToussaint** [comment]
That would be awesome!

**Comment by AntoineToussaint** [comment]
Please let me know if you want help with testing. I can try it out as well.

**Comment by martinRenou** [comment]
> Please let me know if you want help with testing. I can try it out as well

That would be great. I only tried the example you provided, it would be great if you could try your notebook.

You can clone the repository, checkout my branch and install it from sources:
```
git clone https://github.com/jupyter-xeus/xeus-python
cd xeus-python
git fetch https://github.com/martinRenou/xeus-python
git checkout martinRenou/display_module
```
For installing the sources:
https://github.com/jupyter-xeus/xeus-python#installing-from-source

**Comment by AntoineToussaint** [comment]
Will do and let you know.

File: src/xdisplay.cpp [hunk 1]
```diff
@@ -189,6 +286,334 @@ namespace xpyt
...
+     * xdisplay_object declaration *
+     *******************************/
+
+    class xdisplay_object
+    {
+    public:
+
+        xdisplay_object(const py::object& data, const py::object& url, const py::object& filename, const py::object& metadata, const std::string& read_flag = "r");
+        virtual ~xdisplay_object();
+
```
[the remark this answers is not in the document]
Comment (martinRenou) [reply src/xdisplay.cpp:299]: > you should add copy constructor, move constructor, copy assign operator and move assign operator as deleted methods as required for classes that implement entity semantics

Mmh. Not sure. Those constructors and operators will not be used. This class is only meant to be exposed to Python, not used C++ side

**Comment by SylvainCorlay** [comment]
Is this ready to go?

**Comment by SylvainCorlay** [comment]
I think that we need to use another SVG than the QuantStack logo.

**Comment by martinRenou** [comment]
> Is this ready to go?

No... The way the metadata are computed is wrong.

> I think that we need to use another SVG than the QuantStack logo.

Ok, I'll find another one

**Comment by martinRenou** [comment]
Changed the logo to the Jupyter one.

The metadata issue is not a regression. So this PR as it is is already an improvement comparing to master.

**Comment by AntoineToussaint** [comment]
Tried to build from source from master but `plotly` stayed invisible.

I am using a somewhat old version of `jupyterlab` (1.2.7) and not 2.0.x.

And `@jupyterlab/plotly-extension@v1.0.0`.

**Comment by AntoineToussaint** [comment]
When running example from #238, I get `TypeError: 'NoneType' object is not iterable`

**Comment by martinRenou** [comment]
@AntoineToussaint I see. I'll fix it.

**Comment by AntoineToussaint** [comment]
@martinRenou Thanks. Just picked master and rebuilt and it works ;)

I am using it in a Docker image that I use with JupyterHub on kubernetes so very cool!

**Comment by martinRenou** [comment]
Great! xeus-python `0.7.0` is being shipped to conda-forge right now so you'll soon be able to use it :)

**Comment by AntoineToussaint** [comment]
I don't mind a bit of `cmake` in my Dockerfile ;)

**Comment by AntoineToussaint** [comment]
@martinRenou Loving it!

**Comment by martinRenou** [comment]
Great! Happy I could help

**Pull request merged**